server/services/netatmo/lib/netatmo.loadDevices.js

```diff
@@ -0,0 +1,43 @@
+const { default: axios } = require('axios');
+const logger = require('../../../utils/logger');
+const { API } = require('./utils/netatmo.constants');
+
+/**
+ * @description Discover Netatmo cloud devices.
+ * @returns {Promise} List of discovered devices.
+ * @example
+ * await loadDevices();
+ */
+async function loadDevices() {
+  try {
+    const responsePage = await axios({
+      url: API.HOMESDATA,
+      method: 'get',
+      headers: { accept: API.HEADER.ACCEPT, Authorization: `Bearer ${this.accessToken}` },
+    });
+    const { body, status } = responsePage.data;
+    const { homes } = body;
+    let listHomeDevices = [];
+    if (status === 'ok') {
+      const devicesPromises = homes.map(async (home) => {
+        const { modules } = home;
+        if (modules) {
+          return this.loadDeviceDetails(home);
+        }
+        return undefined;
+      });
+      const results = await Promise.all(devicesPromises);
```

Comment:
You are doing a Promise.all on all devices ? It seems that each loadDeviceDetails call the API, so if the user has 50 Netatmo devices, it'll start 50 API calls in parallel, it's way to heavy!

Consider using bluebird Promise.map instead so you can specify a max concurrency :) 

```
const results = await Promise.map(homes, async(home) => 
... 
}, {concurrency: 2} );
```

Reply:
I'm understand, There is normally not too much risk in this regard because the requests are unique for each house. It's very unlikely that a user will have more than 5 houses so I didn't do too much research. But this way it's much cleaner. THANKS.

Reply:
I'm understand, There is normally not too much risk in this regard because the requests are unique for each house. It's very unlikely that a user will have more than 5 houses so I didn't do too much research. But this way it's much cleaner. THANKS.